File: backend/python/app/routes/execute.py
from fastapi import APIRouter

from app.models.execution_result import ExecutionRequest, ExecutionResult
from app.services.sandbox import execute_code

router = APIRouter()


@router.post("/execute", response_model=ExecutionResult)
async def run_code(request: ExecutionRequest):
    return await execute_code(request.code, request.kata_id)

File: backend/rust/src/routes/execute.rs
use axum::Json;

use crate::config::Settings;
use crate::models::execution::{ExecutionRequest, ExecutionResult};
use crate::services::sandbox;

pub async fn execute_code(Json(req): Json<ExecutionRequest>) -> Json<ExecutionResult> {
    let settings = Settings::from_env();
    let result = sandbox::execute_code(&req.code, settings.sandbox_timeout_seconds);
    Json(result)
}
